web3c/eth.js

```diff
@@ -0,0 +1,57 @@
+const ContractFactory = require('./contract_factory');
+const DeployHeader = require('./deploy_header');
+
+/**
+ * Patched web3.eth namespace to provided extended functionality, such as the ability
+ * to specify the contract deploy header on web3.eth.Contract objects.
+ */
+class Eth {
+  constructor(web3) {
+    Object.assign(this, web3.eth);
+    this.__proto__ = web3.__proto__;
+    this.Contract = ContractFactory.make(web3, (_options) => {
+      return new EthContractProvider(web3._requestManager);
+    });
+  }
+}
+
+class EthContractProvider {
```

Comment:
are we going to need to wrap accounts on this object as well?

Reply:
Good point. We should.

Reply:
Done.